src/ff7/widescreen.cpp

```diff
@@ -230,4 +248,67 @@ void ff7_widescreen_hook_init() {
     patch_code_dword(ff7_externals.menu_submit_draw_fade_quad_6CD64E + 0x163, (uint32_t)&wide_viewport_x);
     patch_code_dword(ff7_externals.menu_submit_draw_fade_quad_6CD64E + 0x111, (uint32_t)&wide_viewport_width);
     patch_code_dword(ff7_externals.menu_submit_draw_fade_quad_6CD64E + 0x16F, (uint32_t)&wide_viewport_width);
+}
+
+void Widescreen::loadConfig()
+{
+    char _fullpath[MAX_PATH];
+    sprintf(_fullpath, "%s/%s/config.toml", basedir, external_widescreen_path.c_str());
+
+    try
+    {
+        config = toml::parse_file(_fullpath);
+    }
+    catch (const toml::parse_error &err)
+    {
+        config = toml::parse("");
+    }
+}
+
+void Widescreen::init()
+{
+    loadConfig();
+}
+
+void Widescreen::initParamsFromConfig()
+{
+    static WORD last_field_id = 0;
+
+    struct game_mode* mode = getmode_cached();
+    if (mode->driver_mode != MODE_FIELD) return;
+
+	if (last_field_id != *ff7_externals.field_id)
+	{
+		last_field_id = *ff7_externals.field_id;
+    }
+    else return;
+
+    field_trigger_header* field_triggers_header_ptr = *ff7_externals.field_triggers_header;
+    camera_range.left = field_triggers_header_ptr->camera_range.left;
+    camera_range.right = field_triggers_header_ptr->camera_range.right;
+    camera_range.bottom = field_triggers_header_ptr->camera_range.bottom;
+    camera_range.top = field_triggers_header_ptr->camera_range.top;
+    widescreen_mode = WM_DISABLED;
+
+    auto pName = get_current_field_name();
+    if(pName == 0) return;
+
+    std::string _name(pName);
+    auto node = config[_name];
+    if(node)
+    {
+        camera_range.left = node["left"].value_or(0);
+        camera_range.right = node["right"].value_or(0);
+        camera_range.bottom = node["bottom"].value_or(0);
+        camera_range.top = node["top"].value_or(0);
```

Comment:
You are overwriting all camera ranges. So if there is no config.toml, it's not gonna work at all (everything becomes 0). I think it's better to overwrite only the camera ranges that are inside the toml file.

Reply:
Well in this case if there's or not a config toml, the `if` guard will definitely not allow the code to be executed. Although, if there's at least one node defined, but only with one property, for eg.:
```toml
[field_name]
left = 123
```
In this case all the other fields will be filled with 0, even though they were not defined.

If this is the intended behavior, fine by me, if not, please consider checking if the field exists while reading it to know if you have to override it or not.

Reply:
Added the `if` tests to check whether each of the parameters exists or not.